packages/connection-storage/src/connection-storage.ts

```diff
@@ -55,6 +55,7 @@ const ConnectionSchema: z.Schema<ConnectionWithLegacyProps> = z
           .optional()
           .transform((x) => (x !== undefined ? new Date(x) : x)),
         favorite: z.any().optional(),
+        savedConnectionType: z.enum(['favorite', 'recent']).optional(),
```

Comment:
this is confusing, why do we still have `recents` and `favorites`, based on the new design there are only connections, all of them have a new, some have color, and some of them have `favorite: true`, why do we need "recent", and why do we need to have a `savedConnectionType`?

Reply:
We had another ConnectionInfo that was a bit different, without the savedConnectionType, the favorite field flatten in the parent and a `userFavorite: true`. You can see the commit [here](https://github.com/mongodb-js/compass/pull/5446/commits/1c1e115bc6e5409049c633d39a6b5bfa6567258f#diff-b606feb2d0a7d93ff65ed72b8d0cbb2c355b648e92835e7fd0c17a18b22e6d48).  However, after discussing with Anna, we though that it would be easier if we keep the favorite field but we add a marker field to specify which type of connection type we saved.

The main reason is that before this PR, Compass assumes that every connection with the `favorite` field is a favorite connection, and the `favorite` field is the place where we have the name and the colour. So Compass would use the `savedConnectionType`to categorise. However, it's relevant to say that this is decided only at the ConnectionProvider level. No one outside that knows how the type of the connection is decided (that's why we have two methods). 

And about `recent`, it's because it was saved and the reason was that the user recently connected there. It doesn't have any really specific meaning with history or nothing (which I know it could be confusing after our conversations haha).

Reply:
it is confusing cause `favorite` is an alternative to `recent`. We can keep a backward compatible behavior without saving that information.

Reply:
We discussed it and we will do this factor in another task, due to it's complexity: COMPASS-7672